src/main/java/org/walkmod/javalang/compiler/types/TypesLoaderVisitor.java

```diff
@@ -135,22 +135,27 @@ public TypesLoaderVisitor(SymbolTable symbolTable, SymbolActionProvider actionPr
     * @param imported
     *           if it appears as an import declaration. It is important, because the simple name is
     *           the part after the $ symbol if it is an inner class.
+    * @param importedName
+    *           the name of the import to resolve nested classes against (optional)
     * @return the simple name to be pushed into the symbol table
     */
-   private String getSymbolName(String name, boolean imported) {
+   //@VisibleForTesting
+   static String registrationSymbolName(final String name, final boolean imported, String importedName) {
```

Comment:
I don't like tha name, too. But it's not a simple symbol name but a specially generated one for special purpose of registration. "prepareSymbolName" ? Makes a difference to simple get* and details in javadoc?